knotx-core/src/main/java/com/cognifide/knotx/handler/IncomingRequestsHandler.java

```diff
@@ -50,12 +51,16 @@ public void handle(HttpServerRequest request) {
         try {
             final URI requestUri = new URI(request.path());
             if (repository.support(requestUri)) {
-                repository.get(requestUri, event -> {
+                repository.get(requestUri, request.headers(), event -> {
                     final Template<String, URI> result = event.result();
                     if (result != null) {
+
+                        Action process = actionFactory.newInstance(request);
+                        result.handle(request.response(), process, template -> {
+                            throw new RuntimeException("Can't obtain template!", event.cause());
```

Comment:
Throwing `RTE` here is not good idea. E.g. when we get 302 status code from `HttpTemplate`, `RuntimeException` is rather not expected behaviour?
Pleas just replace it with logger.